src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -316,6 +317,24 @@ public static void DetectUnsupportedModuleParameterAssignments(SemanticModel sem
             }
         }
 
+        public static void DetectInvalidValueForParentProperty(SemanticModel semanticModel, IDiagnosticWriter diagnosticWriter)
+        {
+            foreach (var resourceDeclarationSymbol in semanticModel.Root.ResourceDeclarations)
+            {
+                if (resourceDeclarationSymbol.TryGetBodyPropertyValue(LanguageConstants.ResourceParentPropertyName) is { } referenceParentSyntax)
+                {
+                    var (baseSyntax, _) = SyntaxHelper.UnwrapArrayAccessSyntax(referenceParentSyntax);
+
+                    if (semanticModel.ResourceMetadata.TryLookup(referenceParentSyntax) is not { } && !semanticModel.GetTypeInfo(referenceParentSyntax).IsError())
```

Comment:
You need to do the lookup against `baseSyntax` rather than `referenceParentSyntax` here - I think that's causing the test failures.

For example, if parent is set to `myRes[4]`:
* `referenceParentSyntax` will be `myRes[4]`
* `baseSyntax` will be `myRes`